Uses python client to call OpenStack API

muroi opened the pull request:
The patch replaces curl command used for API call with python-novaclient and -keystoneclient.
It's related to #19.

muroi committed:
Move instance show API to novaclient

Change _do_instance_show() method to use novaclient library.

muroi committed:
Move Instance Stop API to novaclient

Changing call instance stop API by curl command to by novaclient.

muroi committed:
Move Instance Start API to novaclient

Changing calling instance start API by curl command to by novaclient.

muroi committed:
Move Instance Reset Status API to novaclient

Changind do_instance_reset() by curl command to by novaclient.

muroi committed:
Move get server instance list on a hypervisor to novaclient

Using novaclint to get server list on a hypervisor list.

muroi committed:
Move do_host_maintenance_mode to use novaclient

Using novaclient instead of curl command.

muroi committed:
Move evacuate API to novaclient

Using novaclient instead of curl command for instance evauate

muroi committed:
Remove unnecessary method from masakari_utils.py

Some methods for curl command is now unnecessary because of replacing
novaclient.

muroi committed:
Changes UT related to nova_client

sampathP commented on masakari-controller/controller/masakari_util.py:
```diff
@@ -507,722 +511,202 @@ class RecoveryControllerUtilApi(object):
-            self.rc_util.syslogout(msg, syslog.LOG_ERR)
+        msg = ("Project name: %s doesn't exist in project list."
+               % self.rc_config.conf_nova['project_name'])
+        assert len(projects) == 1, msg
```
About the assert statement, this will raise AssertionError. This could happen if you set wrong project_name in config.
IMO, its better to raise smt such as KeyError. Plus, we might loose this logic if run with the -O flag.

muroi replied:
Thank you for reviewing.

I didn't know it would be loose if we run with -O. I'll update it with if-statement.

muroi committed:
Use if-statement to check project's number

assert-statement would be loose if we run it with -O flag. This
patch replaces assert with if in masakari_util.py.

sampathP commented:
@muroi Thank you for the fix.
